src/main/java/org/owasp/benchmark/score/parsers/WebInspectReader.java

```diff
@@ -0,0 +1,154 @@
+/**
+* OWASP Benchmark Project
+*
+* This file is part of the Open Web Application Security Project (OWASP)
+* Benchmark Project For details, please see
+* <a href="https://www.owasp.org/index.php/Benchmark">https://www.owasp.org/index.php/Benchmark</a>.
+*
+* The OWASP Benchmark is free software: you can redistribute it and/or modify it under the terms
+* of the GNU General Public License as published by the Free Software Foundation, version 2.
+*
+* The OWASP Benchmark is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+* even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details
+*
+* @author Dave Wichers <a href="https://www.aspectsecurity.com">Aspect Security</a>
+* @created 2015
+*/
+
+package org.owasp.benchmark.score.parsers;
+
+import java.util.List;
+
+import org.owasp.benchmark.score.BenchmarkScore;
+import org.w3c.dom.Node;
+
+public class WebInspectReader extends Reader {
+	
+	public TestResults parse( Node root ) throws Exception {
+        TestResults tr = new TestResults( "HP WebInspect", true, TestResults.ToolType.DAST);
+
+        // <Scan><Name>Site: https://10.240.28.203:8443/benchmark/</Name><PolicyName>Standard</PolicyName>
+        // <StartTime>9/11/2015 1:56:13 PM</StartTime><Duration>02:59:39.0365257</Duration><Issues><Issue>
+                
+        String duration = getNamedChild("Duration", root ).getTextContent();
+        duration = duration.substring( 0, duration.indexOf( '.' ));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

If the indexOf() search on line 35 does not find the substring and returns -1, the call to substring() on line 35 will throw IndexOutOfBoundsException.